gen_initramfs.sh

```diff
@@ -100,7 +100,13 @@ append_base_layout() {
 	date -u '+%Y%m%d-%H%M%S' > ${TEMP}/initramfs-base-temp/etc/build_date
 	echo "Genkernel $GK_V" > ${TEMP}/initramfs-base-temp/etc/build_id
 
-	printf "$(hostid | sed 's/\([0-9A-F]\{2\}\)/\\x\1/gI')" > ${TEMP}/initramfs-base-temp/etc/hostid
+	little_endian="$(echo I | tr -d [:space:] | od -to2 | head -n1 | awk '{print $2}' | cut -c6)"
```

Comment:
this is more complicated than it needs to be.  you can use `file` on one of the installed files (like `/bin/sh`) to see whether it's big or little endian.